src/frontend/src/optimizer/plan_node/stream_sink.rs

```diff
@@ -510,11 +510,17 @@ impl StreamSink {
     /// The table schema is: | epoch | seq id | row op | sink columns |
     /// Pk is: | epoch | seq id |
     fn infer_kv_log_store_table_catalog(&self) -> TableCatalog {
+        Self::infer_kv_log_store_table_catalog_inner(&self.input, &self.sink_desc().columns)
+    }
+
+    pub fn infer_kv_log_store_table_catalog_inner(
```

Comment:
Let's move this method out of `StreamSink` (maybe into `util`) given that it is not just used by sink.